sdk/src/main/scala/com/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -142,50 +142,38 @@ class AccountForgeMessageBuilder(mainchainSynchronizer: MainchainSynchronizer,
   }
 
   // TODO the stateRoot of the genesis block
-  def getGenesisBlockRootHash =
+  val getGenesisBlockRootHash =
     new Array[Byte](32)
 
-  def getStateRoot(history: AccountHistory, nextConsensusEpochNumber: ConsensusEpochNumber) : (ConsensusEpochNumber, Array[Byte]) = {
+  def getStateRoot( history: AccountHistory,
+                    nextConsensusEpochNumber: ConsensusEpochNumber,
+                    branchPointInfo: BranchPointInfo): Array[Byte] = {
 
     // For given epoch N we should get data from the ending of the epoch N-2.
     // genesis block is the single and the last block of epoch 1 - that is a special case:
     // Data from epoch 1 is also valid for epoch 2, so for epoch N==2, we should get info from epoch 1.
     nextConsensusEpochNumber match {
       case epoch if (epoch <= 2) =>
-        (ConsensusEpochNumber @@ 1, getGenesisBlockRootHash)
+        getGenesisBlockRootHash
 
-      case epoch =>
+      case _ =>
+        val parentId = branchPointInfo.branchPointId
+        val ts = history.blockInfoById(parentId).timestamp
```

Comment:
here we should take a timestamp correlated with `nextConsensusEpochNumber`, not with the parent.
So, replace `nextConsensusEpochNumber` with candidate `timestamp` and use it here.

Reply:
Used timestamp of block to be forged